### PR TITLE
Setup connector for vaultT4

### PR DESCRIPTION
**Pull request by shriyatyagii**

### CODE REVIEW

File: contracts/mainnet/connectors/fluid-vaults/main.sol [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+        address vaultAddress_,
+        uint256 nftId_,
+        int256 newColToken0_,
+        int256 newColToken1_,
+        int256 colSharesMinMax_,
+        int256 newDebtToken0_,
+        int256 newDebtToken1_,
+        int256 debtSharesMinMax_,
+        uint256 repayApproveAmtToken0_,
+        uint256 repayApproveAmtToken1_,
```
Comment (KaymasJain) [inline contracts/mainnet/connectors/fluid-vaults/main.sol:54]: Not needed. If `newDebtToken0_` & `newDebtToken1_` is < 0 then that means payback and approve amount should also be this amount but in uint (not int)

File: contracts/mainnet/connectors/fluid-vaults/main.sol [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+            newDebtToken0_,
+            newDebtToken1_,
+            debtSharesMinMax_,
+            address(this)
+        );
+
+        setUint(setIds_[0], nftId_);
+
+        _setIds(setIds_[1], setIds_[3], uint256(newColToken0_));
+        _setIds(setIds_[2], setIds_[4], uint256(newColToken1_)); // TODO: what if max withdraw?
```
Comment (KaymasJain) [inline contracts/mainnet/connectors/fluid-vaults/main.sol:156]: only operate perfect can do max withdraw & max payback,

File: contracts/mainnet/connectors/fluid-vaults/main.sol [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+            address(this)
+        );
+
+        setUint(setIds_[0], nftId_);
+
+        _handleOperatePerfectSetIds(r_[2], setIds_[1], setIds_[2]);
+        _handleOperatePerfectSetIds(r_[3], setIds_[3], setIds_[4]);
+        _handleOperatePerfectSetIds(r_[5], setIds_[5], setIds_[6]);
+        _handleOperatePerfectSetIds(r_[6], setIds_[7], setIds_[8]);
+
```
Comment (KaymasJain) [inline contracts/mainnet/connectors/fluid-vaults/main.sol:280]: Would also be nice to make the approval 0 as tokens like USDT starts to fail if approval is not 0.

Comment (KaymasJain) [reply contracts/mainnet/connectors/fluid-vaults/main.sol:280]: This applies on both the functions

File: contracts/mainnet/connectors/fluid-vaults/main.sol (hunk 2 above)
Comment (shriyatyagii) [reply contracts/mainnet/connectors/fluid-vaults/main.sol:156]: Got it

File: contracts/mainnet/connectors/fluid-vaults/main.sol (hunk 1 above)
Comment (shriyatyagii) [reply contracts/mainnet/connectors/fluid-vaults/main.sol:54]: Resolved

File: contracts/mainnet/connectors/fluid-vaults/main.sol (hunk 3 above)
Comment (shriyatyagii) [reply contracts/mainnet/connectors/fluid-vaults/main.sol]: Made approval 0 in operate perfect.

**Comment by shriyatyagii** [comment]
Add connector ->
`FLUID-VAULT-T4-A`: [0x8e390D327D3108EFF56a75Fd7EcE31AB94157175](https://etherscan.io/address/0x8e390D327D3108EFF56a75Fd7EcE31AB94157175#code)

**Pull request merged**